workspaces/src/worker/mod.rs

```diff
@@ -13,6 +14,7 @@ use crate::{Network, Result};
 /// deploying a contract, or interacting with transactions.
 pub struct Worker<T: ?Sized> {
     pub(crate) workspace: Arc<T>,
+    pub(crate) on_transact: Option<GasHook>,
```

Comment:
This is only creating a singular gas hook, and would overwrite the previous hook if multiple `GasMeters` were created. Best to have vector of gas hooks instead